crates/indexer/src/models/token_models/petra_activities.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::schema::petra_activities;
+use crate::util::standardize_address;
+use aptos_api_types::{Transaction as APITransaction, WriteSetChange as APIWriteSetChange};
+use field_count::FieldCount;
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+
+#[derive(Debug, Deserialize, FieldCount, Identifiable, Insertable, Serialize)]
+#[diesel(primary_key(account_address, transaction_version))]
+#[diesel(table_name = petra_activities)]
+pub struct PetraActivity {
+    pub transaction_version: i64,
+    pub account_address: String,
+}
+
+impl PetraActivity {
+    pub fn from_transaction(transaction: &APITransaction) -> Vec<Self> {
+        let mut petra_activities = vec![];
+
+        if let APITransaction::UserTransaction(user_txn) = transaction {
+            let version = user_txn.info.version.0 as i64;
+            let mut addresses = HashSet::new();
+
+            // Add the transaction sender.
+            addresses.insert(user_txn.request.sender.to_string());
+
+            // Add any other accounts referenced in the writeset.
+            for wsc in user_txn.info.changes.iter() {
```

Comment:
I'd say go with events only? Token and coin activities only draw from events. write set may create some weird combos later i'd say.